prow/genjobs/cmd/genjobs/main.go

```diff
@@ -403,6 +469,10 @@ func Main() {
 		cleanOutPath(o, o.output)
 	}
 
+	if o.dryRun {
+		return
```

Comment:
Should this just skip `writeOutFile` but otherwise walk through the files? Why are we aborting here?

Reply:
A true `--dry-run` was not what I was going for here. I can rename this to `--clean-only` and implement a more robust `--dry-run` in a follow-up? My current need is that I want to use the `--clean` flag without writing - and I am optimizing time a bit by returning before walking the filesystem (instead of after).